tutorials/lake/nft-indexer.mdx

```diff
@@ -472,99 +382,165 @@ Let's add the interface for later use somewhere after `interface EventLogData`:
-          ).flat()
-        } else if () {
-
-        }
-      }).flat()
+            )
+           })
+        )
+      })
     }
-  })
 ```
 
-:::note
-
-Checkout how we have used `ParasEventLogData` we have defined earlier.
+A few words about what is going on here. If the Receipt's receiver account name ends with `.paras.near` (e.g. `x.paras.near`) we assume it's from Paras marketplace, so we are changing the corresponding variable.
 
-:::
+After that we iterate over the Events and its `data` using the `ParasEventLogData` we've defined earlier. Collecting a list of objects with the NFTs owner and NFTs links.
 
 Mintbase turn, we hope [Nate](https://twitter.com/nategeier) and his team have [migrated to NEAR Lake Framework](./migrating-to-near-lake-framework) already, saying "Hi!" and crafting the link:
 
 ```ts title=index.ts
-  ...
-        } else if (relevantOutcome.receipt.receiverId.includes(".mintbase")) {
-          return (event.data as MintbaseEventLogData[]).map(data => {
-            const memo = JSON.parse(data.memo)
-            return `https://mintbase.io/thing/${memo["meta_id"]}:${relevantOutcome.receipt.receiverId}`
-          })
-        }
-      }).flat()
+  } else if (relevantOutcome.receipt.receiverId.match(/\.mintbase\d+\.near$/)) {
+      marketplace = "Mintbase"
+      nfts = relevantOutcome.events.flatMap(event => {
+        return (event.data as MintbaseEventLogData[])
+          .map(eventData => {
+          const memo = JSON.parse(eventData.memo)
+          return {
+            owner: eventData.owner_id,
+            links: [`https://mintbase.io/thing/${memo["meta_id"]}:${relevantOutcome.receipt.receiverId}`]
+          }
+        })
+      })
     }
+```
+
+Almost the same story as with Paras, but a little bit more complex. The nature of Mintbase marketplace is that it's not a single marketplace! Every Mintbase user has their own store and a separate contract. And it looks like those contract addresses follow the same principle they end with `.mintbaseN.near` where `N` is number (e.g. `nate.mintbase1.near`).
+
+After we have defined that the ExecutionOutcome receiver is from Mintbase we are doing the same stuff as with Paras:
+
+1. Changing the `marketplace` variable
+2. Collecting the list of NFTs with owner and crafted links
+
+And if we can't determine the marketplace, we still want to return something, so let's return Events data as is:
+
+```ts title=index.ts
+  } else {
+    nfts = relevantOutcome.events.flatMap(event => event.data)
+  }
+```
+
+It's time to push the collected data to the `output`
+
+```ts title=index.ts
+  output.push({
+    receiptId: relevantOutcome.receipt.id,
+    marketplace,
+    nfts,
   })
 ```
 
-And if we can't determine the marketplace:
+And make it print the output to the terminal:
 
 ```ts title=index.ts
-        } else {
-          return []
-        }
+if (output.length) {
+  console.log(`We caught freshly minted NFTs!`)
+  console.dir(output, { depth: 5 })
+}
 ```
```

Comment:
Do you believe there is a need in collecting the list before printing out? In my original proposal I switched from `.map()` to `for` loop to separate the collection stage from the printing stage, so `for` loop printed the NFT events immediately without collecting them.

Reply:
I believe this will allow the developers to take this example and:

- send the collected data somewhere
- store it in a database
- send to the websocket

IMO it's more comfortable that the data is grouped.